pkg/distributors/contracts/MultiDistributor.sol

```diff
@@ -392,7 +392,7 @@ contract MultiDistributor is IMultiDistributor, ReentrancyGuard, MultiDistributo
         bytes32 r,
         bytes32 s
     ) external override nonReentrant {
-        IERC20Permit(address(stakingToken)).permit(user, address(this), amount, deadline, v, r, s);
+        IERC20Permit(address(stakingToken)).permit(user, address(getVault()), amount, deadline, v, r, s);
```

Comment:
Just realised that this opens up a troll vector. If people want to give approvals to the vault for other reasons then someone can decide to stake those tokens for them instead.

We should switch this over to using the method to be created for autostaking as then we can rely on all approvals to represent a desire to stake those tokens.